spyder/app/tests/test_mainwindow.py

```diff
@@ -105,6 +105,49 @@ def close_window():
 # Tests
 #==============================================================================
 @flaky(max_runs=3)
+def test_change_cwd_ipython_console(main_window, qtbot, tmpdir):
+    """
+    Test synchronization with working directory and File Explorer when changing Ipython console.
+    """
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
+
+    # Change directory in ipython console using %cd
+    temp_dir = str(tmpdir.mkdir("test_dir"))
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("%cd {}".format(temp_dir))
+    qtbot.wait(1000)
+
+    # assert that cwd changed in workingdirectory
+    assert osp.normpath(main_window.workingdirectory.history[-1]) == osp.normpath(temp_dir)
+
+    # assert that cwd changed in explorer
+    assert osp.normpath(main_window.explorer.treewidget.get_current_folder()) == osp.normpath(temp_dir)
+
+
+@flaky(max_runs=3)
+def test_change_cwd_explorer(main_window, qtbot, tmpdir):
+    """
+    Test synchronization with working directory and Ipython console when changing File Explorer.
```

Comment:
Ipython -> IPython

... when changing *directories* in the File Explorer.

Also, makes this one in two lines too.